packages/frontend/src/pages/Token/LaunchForm/Confirm/Ekubo.tsx

```diff
@@ -48,106 +49,51 @@ export default function EkuboLaunch({ previous }: LastFormPageProps) {
       .multiply(teamAllocationTotalPercentage.multiply(parseFormatedPercentage(ekuboFees).add(1)))
   }, [quoteTokenPrice, startingMcap, teamAllocationTotalPercentage, ekuboFees])
 
-  // starting tick
-  const i129StartingTick = useMemo(() => {
-    if (!quoteTokenPrice || !startingMcap || !memecoin) return
-
-    // initial price in quote/MEME = mcap / quote token price / total supply
-    const initalPrice = +new Fraction(parseFormatedAmount(startingMcap))
-      .divide(quoteTokenPrice)
-      .multiply(decimalsScale(DECIMALS))
-      .divide(new Fraction(memecoin.totalSupply))
-      .toFixed(DECIMALS)
-
-    const startingTickMag = getStartingTick(initalPrice)
-
-    return {
-      mag: Math.abs(startingTickMag),
-      sign: startingTickMag < 0,
-    }
-  }, [quoteTokenPrice, startingMcap, memecoin])
-
-  // fees
-  const fees = useMemo(() => {
-    if (!ekuboFees) return
-
-    return parseFormatedPercentage(ekuboFees).multiply(EKUBO_FEES_MULTIPLICATOR).quotient.toString()
-  }, [ekuboFees])
-
-  // starknet
-  const chainId = useChainId()
-
   // transaction
   const executeTransaction = useExecuteTransaction()
 
   // launch
-  const launch = useCallback(() => {
-    if (!i129StartingTick || !fees || !chainId || !hodlLimit || !memecoin?.address || !teamAllocationQuoteAmount) return
-
-    const uin256TeamAllocationQuoteAmount = uint256.bnToUint256(
-      BigInt(teamAllocationQuoteAmount.multiply(decimalsScale(DECIMALS)).quotient.toString()),
-    )
-
-    const transferCalldata = CallData.compile([
-      FACTORY_ADDRESSES[chainId], // recipient
-      uin256TeamAllocationQuoteAmount, // amount
-    ])
-
-    // team allocation
-    const initalHolders = Object.values(teamAllocation)
-      .filter(Boolean)
-      .map((holder) => holder.address)
-    const initalHoldersAmounts = Object.values(teamAllocation)
-      .filter(Boolean)
-      .map((holder) => uint256.bnToUint256(BigInt(parseFormatedAmount(holder.amount)) * BigInt(decimalsScale(18))))
-
-    // prepare calldata
-    const launchCalldata = CallData.compile([
-      memecoin.address, // memecoin address
-      antiBotPeriod * 60, // anti bot period in seconds
-      +hodlLimit * 100, // hodl limit
-      quoteTokenAddress, // quote token
-      initalHolders, // initial holders
-      initalHoldersAmounts, // intial holders amounts
-
-      fees, // ekubo fees
-      EKUBO_TICK_SPACING, // tick spacing
-      i129StartingTick, // starting tick
-      EKUBO_BOUND, // bound,
-    ])
+  const launch = useCallback(async () => {
+    if (!memecoin || !startingMcap || !quoteToken || !hodlLimit || !ekuboFees) return
+
+    // Sort by index and map to address and parsed amount
+    const teamAllocations = Object.entries(teamAllocation)
+      .sort(([a], [b]) => +a - +b)
+      .map(([, holder]) => ({
+        address: holder.address,
+        amount: parseFormatedAmount(holder.amount),
+      }))
```

Comment:
why not just:

```ts
Object.values(teamAllocation)
      .filter(Boolean)
      .map((holder) => ({
        address: holder.address,
        amount: parseFormatedAmount(holder.amount),
      }))
```

?

Reply:
I thought team allocation index matters, so sorted by it. Removed the sort and map directly.